src/InputManager.cpp

```diff
@@ -174,9 +175,25 @@ void InputManager::KeyInput( const int key, const int action ) {
 }
 
 void InputManager::ScrollWheel( double xoffset, double yoffset ) {
-    if ( yoffset == 1 ) {
-        camera->Zoom( true );
-    } else if ( yoffset == -1 ) {
-        camera->Zoom( false );
+    if ( scene->AnyGeometrySelected() == true) {
+        for ( int i = 0; i < scene->GetGeometriesSize(); i++ ) {
+            Geometry* geometry = scene->GetGeometry( i );
+
+            if ( !geometry->IsSelectedGeometry() ) {
+                continue;
+            }
+
+            if ( yoffset == 1 ) {
+                geometry->ModelMatrix = glm::scale( geometry->ModelMatrix, glm::vec3( GEOMETRY_SCALE_UP ) ) ;
+            } else if ( yoffset == -1 ) {
+                geometry->ModelMatrix = glm::scale( geometry->ModelMatrix, glm::vec3( GEOMETRY_SCALE_DOWN ) );
+            }
+        }
+    } else {
+        if ( yoffset == 1 ) {
+            camera->Zoom( true );
+        } else if ( yoffset == -1 ) {
+            camera->Zoom( false );
```

Comment:
you can get rid of this inner if-else by doing:
`camera->Zoom( yoffset == 1 );`